calculation_scripts/OptimizationLoop.py

```diff
@@ -0,0 +1,167 @@
+
+    # Geometry parameters.
+    geometry_config = np.loadtxt("geometry.txt", dtype=int)
+    geo_indices = geometry_config[4:]
+    geo_nx, geo_ny, geo_nz, geo_ntotal = geometry_config[:4]
+    geo_pixel_size_nm = 15.0
+
+    # Dielectric (optimizable value) parameters.
+    dielectric_values = np.loadtxt("hourglass.txt")
+    #dielectric_values += np.random.uniform(0, 10e-6, size=dielectric_values.shape)
+    dielectric_values = np.clip(dielectric_values, 0, 1)
+    dielectric_materials = [1.01 + 0j, 5.96282 + 3.80423e-7j]
+
+    # Incident light wave parameters.
+    light_direction = [0, 0, 1]
+    light_polarization = [1, 0, 0]
+    light_amplitude = 1.0
+    light_wavelength_nm = 542
+    background_refractive_index = np.sqrt(
+        np.real(dielectric_materials[0]))
+
+    # Numerical optimization parameters.
+    # A-product-core periodicity parameters / boundaries.
+    apc_max_m = 50
+    apc_max_n = 50
+    apc_l_m = 22
+    apc_l_n = 22
+    apc_method_name = "FCD"
+
+    # Cached integral values.
+    sici_delta = 0.1
+    sici_n = 1_000_000
+    integral_positions = np.linspace(sici_delta, sici_delta * sici_n, sici_n - 1)
+    si, ci = sici(integral_positions)
+
+    ordered_parameters = {
+        # Filter parameters.
+        "filter_beta_min": filter_beta_min,
+        "filter_beta_max": filter_beta_max,
+        "filter_ita": filter_ita,
+        "filter_method": filter_method,
+        "filter_iterations": filter_iterations,
+        "filter_radii": filter_radii,
+        "filter_enable": filter_enable,
+        # Symmetry parameters.
+        "sym_method": sym_method,
+        "sym_axis": sym_axis,
+        "sym_is_periodic": sym_is_periodic,
+        # Objective and geometry.
+        "obj_name": obj_name,
+        "obj_config": obj_config,
+        "geo_indices": geo_indices,
+        "dielectric_values": dielectric_values,
+        "geo_nx": geo_nx,
+        "geo_ny": geo_ny,
+        "geo_nz": geo_nz,
+        "geo_ntotal": geo_ntotal,
+        # Incident light wave (plus some material properties).
+        "light_direction": light_direction,
+        "light_amplitude": light_amplitude,
+        "light_polarization": light_polarization,
+        "light_wavelength_nm": light_wavelength_nm,
+        "dielectric_materials": dielectric_materials,
+        "background_refractive_index": background_refractive_index,
+        # Numerical parameters.
+        "apc_max_m": apc_max_m,
+        "apc_max_n": apc_max_n,
+        "apc_l_m": apc_l_m,
+        "apc_l_n": apc_l_n,
+        "apc_method_name": apc_method_name,
+        "geo_pixel_size_nm": geo_pixel_size_nm,
+        # Sine and cosine integral values.
+        "sineIntegralValues": si,
+        "cosineIntegralValues": ci,
+        "integralDelta": sici_delta,
+        "verbose": True,
+    }
+
+    model = dda_model.DDAModel(*list(ordered_parameters.values()))
+    return model, ordered_parameters
+
+
+
+GLOBAL_MODEL = _construct_test_model()
```

Comment:
Nit: ALL_CAPS is used for global variables when we expect state to be shared. We did this in the tests because the pytest harness will run each test independently (though we ought to migrate to fixtures, to be "correct"). Here, it's more appropriate to just name this `model` since it'll only be used in this script.

Reply:
good to know! so if a define a variable that will be used in some function later in the same script, it should be capitalized?

Reply:
Not a hard and fast rule. This one is pretty contextual and relies on design patterns we haven't discussed yet, so I would actually tend to avoid ALL_CAPS for now.